src/charm.py

```diff
@@ -230,6 +228,10 @@ def _on_bessd_pebble_ready(self, event: EventBase) -> None:
         """Handle Pebble ready event."""
         if not self.unit.is_leader():
             return
+        if not self._bessd_container.can_connect():
+            self.unit.status = WaitingStatus("Waiting for bessd container to be ready")
+            event.defer()
```

Comment:
Here we can remove this `defer()`, this handler will be called again by `PebbleReady`.